### ISSUE TITLE
Why is the inability to connect to the server getting stored?

### ISSUE THREAD

**Issue by hsyhhssyy** [intersystems-community/vscode-objectscript#1196]
If, while a VSCode window is open, I'm unable to connect to a server due to, for instance, being disconnected from the network or connected to a private network, I would receive an error message when trying to connect. That's correct.

Subsequently, without closing the VSCode window, if I modify my network to enter an environment that can reach to this server and then choose to refresh the connection or select 'Toggle Connection' in the status bar to try to toggle it off and back on, I would still be prompted with an inability to connect. 

Only after restarting the VSCode window am I able to connect to this server.

**Comment by gjsjohnmurray** [comment]
@hsyhhssyy I have been able to reproduce this. The status bar panel is provided by the ObjectScript extension rather than Server Manager (in my repro testing, Server Manager's own tree didn't exhibit the problem, but instead correctly reconnected when refreshed after my VPN to the server's network was re-established).

I will work on a fix now.

**Comment by gjsjohnmurray** [comment]
@hsyhhssyy please download [this zip](https://github.com/intersystems-community/vscode-objectscript/suites/14587924772/artifacts/826416151), extract the VSIX, install it and try to reproduce your issue. Also check that ordinary operation hasn't been broken.

**Comment by hsyhhssyy** [comment]
@gjsjohnmurray 

I greatly appreciate your help. I have installed this plugin, which indeed resolved my problem. Now, toggling off and back on will refresh the connection.

I have tested the following scenarios:

1. When the Workspace has not saved the Server, and the VPN is disconnected, if I launch VSCode, click the "Choose Server and Namespace" button, and select the Server, an error occurs. Then I connect to the VPN, and press the "Choose Server and Namespace" button to select the Server again, the old 2.10 plugin would continue to error out, while the new 2.10.1-dev plugin successfully connects.
2. When the Workspace has saved the Server and the VPN is disconnected, if I start VSCode and attempt to access an object in the "ObjectScript" panel, an error occurs. After connecting to the VPN, then:
    * If I click the button in the Status Bar and select Refresh Connection, the old 2.10 plugin will error out, but the new 2.10.1-dev plugin will successfully connect.
    * If I click the button in the Status Bar and select Toggle connection, the old 2.10 plugin will toggle off, and trying to toggle on will result in an error. With the new 2.10.1-dev plugin, toggling back on successfully connects.

However, I need to continue using it for a while to confirm whether this problem has been completely resolved under other rare situations. For example, if the VPN unexpectedly interrupts during the toggle on process, will this cause any inconsistencies? Such problems are difficult to test directly and can only be evaluated if I accidentally encounter them during use.

**Comment by gjsjohnmurray** [comment]
@hsyhhssyy thanks for testing the PR. I decided to merge it, so the fix is now available in the latest beta VSIX at https://github.com/intersystems-community/vscode-objectscript/releases

If you find edge cases where the problem remains please open a new issue.
